test/test_basic.py

```diff
@@ -94,8 +94,9 @@ async def test_connection_refused(runtime_dir: pathlib.Path) -> None:
 @pytest.mark.asyncio
 async def test_dns_error(runtime_dir: pathlib.Path) -> None:
     session = ferny.Session()
-    with pytest.raises(socket.gaierror):
+    with pytest.raises(Exception) as exc_info:
```

Comment:
The docs look like you can pass a tuple of expected exception types here.

https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest.raises

Reply:
I actually did look at the docs, but I missed the Union(Tuple) thing. Seems to work!